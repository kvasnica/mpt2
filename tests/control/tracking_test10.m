function tracking_test10

% tests tracking=1/2 for MLD systems with no equivalent MLD formulation.
% related to tracking_test11.m

% MLD system with no PWA representation
sysStruct = mpt_sys('pwa_car.hys', 'nopwa');
probStruct.Q = eye(2); 
probStruct.R = 1; 
probStruct.N = 1; 
probStruct.norm = 2;

probStruct.tracking = 1;
ctrl = mpt_control(sysStruct, probStruct, 'online');
[C, O, V] = mpt_ownmpc(sysStruct, probStruct, 'online');
ctrl2 = mpt_ownmpc(sysStruct, probStruct, C, O, V);

probStruct.tracking = 2;
ctrl = mpt_control(sysStruct, probStruct, 'online');
[C, O, V] = mpt_ownmpc(sysStruct, probStruct, 'online');
ctrl2 = mpt_ownmpc(sysStruct, probStruct, C, O, V);
