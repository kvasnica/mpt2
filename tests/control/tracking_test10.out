Value of "probStruct.subopt_lev" not specified, assuming 0...
tracking=1 cannot be used for MLD systems, switching to tracking=2
Using an MLD model...

For tracking to work, you will need to provide
"Options.reference" when calling "u=ctrl(x0, Options)".
tracking=1 cannot be used for MLD systems, switching to tracking=2
Using an MLD model...

For tracking to work, you will need to provide
"Options.reference" when calling "u=ctrl(x0, Options)".
Value of "probStruct.subopt_lev" not specified, assuming 0...
Using an MLD model...

For tracking to work, you will need to provide
"Options.reference" when calling "u=ctrl(x0, Options)".
Using an MLD model...

For tracking to work, you will need to provide
"Options.reference" when calling "u=ctrl(x0, Options)".
